In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('data/income_evaluation.csv')
df.head()

age          workclass   fnlwgt   education   education-num  \
0   39          State-gov    77516   Bachelors              13   
1   50   Self-emp-not-inc    83311   Bachelors              13   
2   38            Private   215646     HS-grad               9   
3   53            Private   234721        11th               7   
4   28            Private   338409   Bachelors              13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

    capital-gain   capital-loss   hours-per-week  native-country  income  
0           2174              0               40   United-States   <=50K  
1              0              0               13   United-States   <=50K  
2              0              0               40   United-States   <=50K  
3              0              0               40   United-States   <=50K  
4              0              0               40            Cuba   <=50K

In [3]:
df.shape

(32561, 15)

In [4]:
df = df.sample(frac=1)

In [31]:
df.head()

age   workclass   fnlwgt      education   education-num  \
1755    25     Private   169323      Bachelors              13   
29885   46     Private   336984        HS-grad               9   
2645    46     Private   138626   Some-college              10   
18893   53   Local-gov   197054        Masters              14   
9468    40   State-gov   199381        Masters              14   

            marital-status          occupation  relationship    race      sex  \
1755    Married-civ-spouse     Exec-managerial     Own-child   White   Female   
29885        Never-married   Handlers-cleaners     Own-child   White     Male   
2645    Married-civ-spouse        Craft-repair       Husband   White     Male   
18893   Married-civ-spouse      Prof-specialty       Husband   White     Male   
9468    Married-civ-spouse      Prof-specialty          Wife   White   Female   

        capital-gain   capital-loss   hours-per-week  native-country  income  
1755               0              0               40   United-States   <=50K  
29885              0              0               17   United-States   <=50K  
2645               0              0               40   United-States   <=50K  
18893              0              0               40   United-States    >50K  
9468           15024              0               37   United-States    >50K

In [6]:
len(df) * 0.8

26048.800000000003

In [7]:
train = df[:26048]

In [8]:
test = df[26048:]

In [9]:
train.head()

age   workclass   fnlwgt      education   education-num  \
1755    25     Private   169323      Bachelors              13   
29885   46     Private   336984        HS-grad               9   
2645    46     Private   138626   Some-college              10   
18893   53   Local-gov   197054        Masters              14   
9468    40   State-gov   199381        Masters              14   

            marital-status          occupation  relationship    race      sex  \
1755    Married-civ-spouse     Exec-managerial     Own-child   White   Female   
29885        Never-married   Handlers-cleaners     Own-child   White     Male   
2645    Married-civ-spouse        Craft-repair       Husband   White     Male   
18893   Married-civ-spouse      Prof-specialty       Husband   White     Male   
9468    Married-civ-spouse      Prof-specialty          Wife   White   Female   

        capital-gain   capital-loss   hours-per-week  native-country  income  
1755               0              0               40   United-States   <=50K  
29885              0              0               17   United-States   <=50K  
2645               0              0               40   United-States   <=50K  
18893              0              0               40   United-States    >50K  
9468           15024              0               37   United-States    >50K

In [10]:
train.columns

Index(['age', ' workclass', ' fnlwgt', ' education', ' education-num',
       ' marital-status', ' occupation', ' relationship', ' race', ' sex',
       ' capital-gain', ' capital-loss', ' hours-per-week', ' native-country',
       ' income'],
      dtype='object')

In [11]:
train.columns = train.columns.str.lstrip()

In [12]:
test.columns = test.columns.str.lstrip()

In [13]:
X_train = train.drop('income', axis=1)

In [14]:
X_train.head()

age   workclass  fnlwgt      education  education-num  \
1755    25     Private  169323      Bachelors             13   
29885   46     Private  336984        HS-grad              9   
2645    46     Private  138626   Some-college             10   
18893   53   Local-gov  197054        Masters             14   
9468    40   State-gov  199381        Masters             14   

            marital-status          occupation relationship    race      sex  \
1755    Married-civ-spouse     Exec-managerial    Own-child   White   Female   
29885        Never-married   Handlers-cleaners    Own-child   White     Male   
2645    Married-civ-spouse        Craft-repair      Husband   White     Male   
18893   Married-civ-spouse      Prof-specialty      Husband   White     Male   
9468    Married-civ-spouse      Prof-specialty         Wife   White   Female   

       capital-gain  capital-loss  hours-per-week  native-country  
1755              0             0              40   United-States  
29885             0             0              17   United-States  
2645              0             0              40   United-States  
18893             0             0              40   United-States  
9468          15024             0              37   United-States

In [15]:
y_train = train.income

In [16]:
y_train.head()

1755      <=50K
29885     <=50K
2645      <=50K
18893      >50K
9468       >50K
Name: income, dtype: object

In [17]:
X_test = test.drop('income', axis='columns')

In [18]:
y_test = test.income

# Using sklearn's train_test_split

In [19]:
d = pd.read_csv('data/income_evaluation.csv')
d.head()

age          workclass   fnlwgt   education   education-num  \
0   39          State-gov    77516   Bachelors              13   
1   50   Self-emp-not-inc    83311   Bachelors              13   
2   38            Private   215646     HS-grad               9   
3   53            Private   234721        11th               7   
4   28            Private   338409   Bachelors              13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

    capital-gain   capital-loss   hours-per-week  native-country  income  
0           2174              0               40   United-States   <=50K  
1              0              0               13   United-States   <=50K  
2              0              0               40   United-States   <=50K  
3              0              0               40   United-States   <=50K  
4              0              0               40            Cuba   <=50K

In [20]:
from sklearn.model_selection import train_test_split

In [21]:
X = d.drop(' income', axis=1)
y = d[' income']

In [22]:
X1, X2, y1, y2 = train_test_split(X, y, test_size=0.2, random_state=0, stratify=y)

In [23]:
y1.value_counts()

 <=50K    19775
 >50K      6273
Name:  income, dtype: int64

In [24]:
y2.value_counts()

 <=50K    4945
 >50K     1568
Name:  income, dtype: int64

In [25]:
y.value_counts()

 <=50K    24720
 >50K      7841
Name:  income, dtype: int64

24720 is the total number of "<=50K" data samples. <br>These will be split in the ratio of 80:20

In the 'training y', i.e. y1, we'll have 24720 * 0.8 = 19776 "<=50K" samples

In [26]:
24720*0.8

19776.0

In the 'testing y', i.e. y2, we'll have 24720 * 0.2 = 4944 "<=50K" samples

In [27]:
24720*0.2

4944.0

7841 is the total number of ">50K" data samples. <br>These will be split in the ratio of 80:20

In the 'testing y', i.e. y2, we'll have 7841 * 0.2 = 1568 ">50K" samples (rounding off)

In [28]:
7841*0.2

1568.2

In the 'training y', i.e. y1, we'll have 7841 * 0.8 = 6273 ">50K" samples (rounding off)

In [29]:
7841*0.8

6272.8

In [30]:
X1.head()

age  workclass   fnlwgt      education   education-num  \
29101   19          ?   140399   Some-college              10   
10606   50    Private   158284   Some-college              10   
9414    62    Private   183735   Some-college              10   
1747    20    Private   154781        HS-grad               9   
10677   25    Private   356344      Bachelors              13   

            marital-status          occupation     relationship    race  \
29101        Never-married                   ?   Other-relative   White   
10606   Married-civ-spouse   Machine-op-inspct          Husband   White   
9414    Married-civ-spouse     Exec-managerial          Husband   White   
1747         Never-married        Adm-clerical    Not-in-family   White   
10677        Never-married        Adm-clerical        Own-child   White   

           sex   capital-gain   capital-loss   hours-per-week  native-country  
29101   Female              0              0               30   United-States  
10606     Male              0              0               40   United-States  
9414      Male              0              0               40   United-States  
1747    Female              0              0               40   United-States  
10677   Female              0              0               40   United-States